xapian-core/weight/lmweight.cc

```diff
@@ -176,8 +176,8 @@ LMWeight::get_sumpart(Xapian::termcount wdf, Xapian::termcount len,
 	 * case query term is present in the document.
 	 */
 	double weight_document = wdf_double / len_double;
-	weight_sum = (param_smoothing1 * weight_collection) +
-		     ((1 - param_smoothing1) * weight_document);
+	weight_sum = 1 + ((((1 - param_smoothing1) / param_smoothing1) * 
+                (weight_document / weight_collection)));
```

Comment:
There seem to be more parentheses here than are needed, which makes the expression harder for the reader to understand.  The opening `((` appear to be closed together by the final `))`.

Also the start of the two arguments of the `*` where the line is split should be aligned vertically (like the two arguments of `+` are in the old version of the code).